internal/characters/childe/abil.go

```diff
@@ -0,0 +1,513 @@
+					if c.rtflashICD[t.Index()] < c.Core.F {
+						c.AddTask(func() {
+							d := c.Snapshot(
+								"Riptide Flash",
+								core.AttackTagNormal,
+								core.ICDTagTartagliaRiptideFlash,
+								core.ICDGroupDefault,
+								core.StrikeTypeDefault,
+								core.Hydro,
+								25,
+								rtFlash[0][c.TalentLvlAttack()],
+							)
+							d.Targets = core.TargetAll
+
+							//proc 3 hits
+							for i := 1; i <= 3; i++ {
+								x := d.Clone()
+								c.QueueDmg(&x, i)
+							}
+							c.Core.Log.Debugw("Riptide Flash ticked", "frame", c.Core.F, "event", core.LogCharacterEvent, "dur",
+								c.Core.Status.Duration("childemelee"), "target", t.Index(), "fl", c.rtflashICD[t.Index()], "sl", c.rtslashICD[t.Index()], "rtExpiry", c.rtExpiry[t.Index()])
+						}, "Riptide Flash", 5)
+						c.rtflashICD[t.Index()] = c.Core.F + 42 //0.7s icd
+					}
+				}
+
+				c.applyRT(t.Index())
+			}
+		case core.AttackTagElementalBurst:
+			if c.Core.Status.Duration("childemelee") > 0 {
+				//Riptide Blast: Clears Riptide status. DMG Dealt is considered Elemental Burst Damage.
+				//clear riptide status
+				if c.rtExpiry[t.Index()] > c.Core.F {
+					// c.Core.Log.Debugw(fmt.Sprintf("%v checking for tick", sname), "frame", c.Core.F, "event", core.LogCharacterEvent, "dur",
+					// 	c.Core.Status.Duration("childemelee"), "target", t, "fl", c.rtflashICD[t], "sl", c.rtslashICD[t], "rtExpiry", c.rtExpiry[t])
+					c.AddTask(func() {
+						d := c.Snapshot(
+							"Riptide Blast",
+							core.AttackTagElementalBurst,
+							core.ICDTagNone,
+							core.ICDGroupDefault,
+							core.StrikeTypeDefault,
+							core.Hydro,
+							50,
+							rtBlast[c.TalentLvlBurst()],
+						)
+						d.Targets = core.TargetAll
+
+						c.Core.Combat.ApplyDamage(&d)
+						c.Core.Log.Debugw("Riptide Blast ticked", "frame", c.Core.F, "event", core.LogCharacterEvent, "dur",
+							c.Core.Status.Duration("childemelee"), "target", t.Index(), "fl", c.rtflashICD[t.Index()], "sl", c.rtslashICD[t.Index()], "rtExpiry", c.rtExpiry[t.Index()])
+					}, "Riptide Blast", 5)
+
+					c.rtExpiry[t.Index()] = 0
+
+					c.Core.Log.Debugw("Childe cleared riptide", "frame", c.Core.F, "event", core.LogCharacterEvent, "dur",
+						c.Core.Status.Duration("childemelee"), "target", t.Index(), "Expiry", c.rtExpiry[t.Index()])
+
+					if c.Base.Cons >= 4 {
+						c.funcC4[t.Index()] = false
+						c.mlBurstUsed = true
+					}
+				}
+			} else {
+				c.applyRT(t.Index())
+			}
+		default:
+		}
+		return false
+	}, "childe-riptide")
+}
+
+//Q: does all type of childe's riptide share the same icd of particle?
+//A: Hard 3s cooldown on particle generation. Particles can only be gained through riptide procs.
+//Q: do all riptide flash, burst, slash and blast share same icd of particle generation?
+//A: flash and slash share it iirc, burst doesn't have one, and I don't think blast gens energy
+func (c *char) rtParticleGen() {
```

Comment:
Maybe I'm missing something, but I'm not sure why this function needs to needs to be a separate event subscription. Can't you just handle riptide energy procs after each riptide instance in the main riptide function or something?